Add a new Module for the Liquidity Pool project(s)

chrislemaire opened the pull request:
In this pull request we propose a module for trading currencies through a liquidity pool.

chrislemaire commented:
Ah finally got it to pass, turns out Intellij tried to give me a new and improved version of some library, but the tests didn't like that...

Tim-W commented on liquidity-pool/build.gradle:
```diff
@@ -0,0 +1,126 @@
+    implementation project(':common')
+    implementation project(':ipv8-android')
+
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
```
Apart from the first two 
```
    implementation project(':common')
    implementation project(':ipv8-android')
```
Please look into what plugins/modules actually need for your project. (You may not need ttorrent, bitcoinj, mp3 library etc)

chrislemaire replied:
We're not exactly sure what we'll need for now, so I'll just strip it to the bare minimum for now then.

chrislemaire replied:
I don't really understand what is making these tests fail now. It seems something is wrong with currencyii, but as far as I can find I didn't change anything regarding it or in its module?

chrislemaire commented:
@Tim-W Could you take another look?

xoriole commented on common/src/main/java/nl/tudelft/trustchain/common/bitcoin/WalletService.kt:
```diff
@@ -0,0 +1,110 @@
+
+    companion object {
+        private lateinit var globalWallet: WalletAppKit
+        private val bitcoinFaucetEndpoint = "http://134.122.59.107:3000"
```
It'd better to extract out this endpoint into config (or even to gradle config) instead of hardcoding here.

kchong97 replied:
I have moved the endpoint to the build.grade

xoriole commented on common/src/main/java/nl/tudelft/trustchain/common/bitcoin/WalletService.kt:
```diff
@@ -0,0 +1,110 @@
+                }
+            }
+
+            app.setPeerNodes(PeerAddress(params, InetAddress.getByName("134.122.59.107"), params.port))
```
Same here

kchong97 replied:
Moved to build.gradle

xoriole reviewed (commented):
Looking at the commit history, there are quite some repeated commits (maybe because of other merges). Could you please rebase the PR and squash appropriate commits to clean the commit history?

devos50 commented:
@xoriole could you please close https://github.com/Tribler/tribler/issues/5988, https://github.com/Tribler/tribler/issues/5996 and https://github.com/Tribler/tribler/issues/6001 after merging this PR? 👍

xoriole commented on liquidity-pool/src/main/java/nl/tudelft/trustchain/liquidity/ui/PoolFragment.kt:
```diff
@@ -0,0 +1,333 @@
+        }
+
+        lifecycleScope.launchWhenStarted {
+            poolBitcoinAddress = sharedPreference.getString("poolBitcoinAddress", "mv7x4cqQMM8ptFgiHwJZTMTQTcTFV5rteU").toString()
```
Please extract out these hardcoded values from here

xoriole commented on liquidity-pool/src/main/java/nl/tudelft/trustchain/liquidity/ui/PoolFragment.kt:
```diff
@@ -0,0 +1,333 @@
+            }
+            /**
+             * if transaction is pending from before, disable button
+             * TODO: Uncomment line below when not testing
```
Please remove commented code

xoriole commented on liquidity-pool/src/main/java/nl/tudelft/trustchain/liquidity/data/EthereumGethMultiSigWallet.kt:
```diff
@@ -0,0 +1,56 @@
+        // if contract already bound, don't do anything
+        if (contractBound) return
+        // else, let's bind!
+        val contractAddressHex = "0x2DfEecF7d2f5363149cC73cAb96C00575c732170"
```
Please extract out this contract address and document it

xoriole commented on liquidity-pool/src/main/java/nl/tudelft/trustchain/liquidity/data/EthereumGethMultiSigWallet.kt:
```diff
@@ -0,0 +1,56 @@
+    init {
+        singleGethWallet = gethWallet
+        gethNode = gethWallet.node
+        // val transactOpts = TransactOpts()
```
Please remove these commented out code

xoriole commented on eurotoken/src/main/java/nl/tudelft/trustchain/eurotoken/ui/settings/DefaultGateway.kt:
```diff
@@ -5,10 +5,11 @@ import nl.tudelft.ipv8.util.hexToBytes
 
 object DefaultGateway {
-    val ip = "gateway.euro-token.nl"
+    val ip = "188.166.94.39" // "gateway.euro-token.nl"
```
Please extract out these variables values in the config.

xoriole reviewed (changes requested):
Nice!

Some comments:
- Please extract out some the mentioned hardcoded values
- Remove commented out code
- Remove .DS_Store and add it to .gitignore

xoriole commented:
@chrislemaire Could you please rebase the PR?

chrislemaire commented:
I rebased and checked the error. Looks like the error has to do with a dependency that we didn't use nor alter in the common module, so this might well be a problem that needs to be fixed in master, rather than this branch. This might understandably take some time. Since this has nothing to do with our project and we have moved on to different courses and activities since then, I propose that someone more actively involved with the superapp makes a copy of this branch to continue the Pull Request.

devos50 commented:
> What went wrong:
  Could not determine the dependencies of task ':app:lint'.
  Could not resolve all artifacts for configuration ':app:releaseRuntimeClasspath'.
  Could not find com.goterl.lazycode:lazysodium-android:4.1.0.

Yes, this is a known issue and not related to your PR (also related to #71). Bintray stopped offering their free service tier and the superapp pulls various of its dependencies from Binary.

thversfelt commented:
I will leave this here, since then all project members and supervisors can see (I don't have all e-mail addresses): I am speaking on behalf of both Liquidity Pool project contributors. We noticed that our project grades our missing on https://my.tudelft.nl/. Would it be possible to add our grades, considering that this PR can basically be merged since the checks fail for reasons unrelated to this PR?

Thank you in advance!

synctext commented:
+6,311 lines of re-basing, that will take our staff a full day or more. OK.
Bintray issue is now documented here, it disrupted our workflow significantly: https://github.com/Tribler/tribler/issues/5914#issuecomment-848537738
_All clouds are fragile_
(all students seemed to have finished, grades will be finalised)
